src/index.js

```diff
@@ -1,32 +1,101 @@
-      else return callback(url_api);
+// Como la variable se declaró en mayúsculas, asumo que no cambiará
+// por tanto la decararé como constante ya que las constantes
+// por convención se nombran en mayúculas.
+const API = 'https://rickandmortyapi.com/api/character/';
+
+
+
+
+// ------------- Eliminando callback hell ---------------------
+
+// EN ES6 la mejor manera de evitar el callback hell
+// es mediante el uso de promesas
+// En esta ocasió fetch data es una función que no responde
+// de manera síncrona, ya que depende del tiempo de respuesta del API
+// por tanto cambié su funcionamiento para que retorne una promesa
+function fetchData (url_api) {
+  
+  return new Promise( (resolve, reject) => {
+
+    // Dejo de usar el xhttp como variable gobal
+    // así mejora la seguridad de la aplicación
+    // y se vuelve una función fácil de reutilizar
+    let XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+    let req = new XMLHttpRequest();
+
+  
+    req.onreadystatechange = (event) => {
+      if(req.readyState === 4){
+        if(req.status == 200){
+          resolve(JSON.parse(req.responseText))
+        }else{
+          // Manejando el error de todos los callbacks
+          reject(`Error: ${req.status}, No hay resultados :(`)
+        }
+      }
     }
-  };
-  xhttp.open('GET', url_api, false);
-  xhttp.send();
-};
-
-fetchData(API, function (error1, data1) {
-  if (error1) return console.error('Error' + ' ' + error1);
-  console.log('Primer Llamado...')
-  fetchData(API + data1.results[0].id, function (error2, data2) {
-    if (error2) return console.error(error1);
+
+    req.open('GET', url_api, false);
+    req.send(null);
+
+  } )  
+
+}
+
+async function fullRequest(api) {
+ 
+  // De acuerdo con lo aprendido tras mi pregunta sobre try catch
+  // David Aoresti mencionó que es buena práctica usar un try por cada
+  // tipo de error similar.
+
+  // En este caso todas las promesas darán un mismo tip de error
+  // puesto que llaman a la misma función
+  try{
+    console.log('Primer Llamado...')
+    let data1 = await fetchData(api)
+    
     console.log('Segundo Llamado...')
-    fetchData(data2.origin.url, function (error3, data3) {
-      if (error3) return console.error(error3);
-      console.log('Tercero Llamado...')
-      console.log('Personajes:' + ' ' + data1.info.count);
-      console.log('Primer Personaje:' + ' ' + data2.name);
-      console.log('Dimensión:' + ' ' + data3.dimension);
-    });
-  });
-});
\ No newline at end of file
+    let data2 = await fetchData(api + data1.results[0].id)
+
+    console.log('Tercer Llamado...')
+    let data3 = await fetchData(data2.origin.url)
+
+    console.log(`Personajes ${data1.info.count}`)
+    console.log(`Primer Personaje: ${data2.name}`)
+    console.log(`Dimensión: ${data3.dimension}`)
+
+  }catch(error){
+    console.log(error)
+  }
+  
+}
+
+
+// Mandamos a llamar a la función que obtiene toda la información del show
+fullRequest(API)
```

Comment:
Aveces pones ; otras veces no, hay varios estandares, puedes elegir uno donde si pones ; o donde no lo pones.

Investiga: JavaScript Standard 

```suggestion
fullRequest(API);
```